Sources/CoreAardvark/Logging/ARKDataArchive.m

```diff
@@ -199,6 +199,19 @@ - (void)saveArchiveAndWait:(BOOL)wait;
     }
 }
 
+- (void)saveArchiveWithCompletionHandler:(nullable dispatch_block_t)completionHandler;
+{
+    NSBlockOperation *completionOperation = [NSBlockOperation blockOperationWithBlock:^{
+        [self.fileHandle synchronizeFile];
+
+        if (completionHandler != NULL) {
+            completionHandler();
+        }
+    }];
+    completionOperation.qualityOfService = NSQualityOfServiceUserInitiated;
```

Comment:
This a pretty intense QoS for something that's invisible to a customer. Might be good to add a comment as to why we're utilizing this QOS.

Reply:
Addressed in 0ad0e56.